src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,665 @@
+    software : str
+        The software used for generating input. {'GROMACS', 'AMBER'}
+    dir : str
+        Directory in which data files are stored. Default: os.path.curdir.
+    prefix : str
+        Prefix for datafile sets. Default: 'dhdl'.
+    suffix : str
+        Suffix for datafile sets. Default: 'xvg'.
+    T : float
+        Temperature in K. Default: 298.
+    outdirectory : str
+        Directory in which the output files produced by this script will be
+        stored. Default: os.path.curdir.
+
+    Attributes
+    ----------
+    logger : Logger
+        The logging object.
+    file_list : list
+        The list of filenames sorted by the lambda state.
+    '''
+    def __init__(self, units='kT', software='Gromacs', dir=os.path.curdir,
+                 prefix='dhdl', suffix='xvg', T=298, outdirectory=os.path.curdir):
+
+        super().__init__(units, software, T, outdirectory)
+        self.logger = logging.getLogger('alchemlyb.workflows.ABFE')
+        self.logger.info('Initialise Alchemlyb ABFE Workflow')
+        self.logger.info(f'Alchemlyb Version: f{__version__}')
+        self.logger.info(f'Set Temperature to {T} K.')
+        self.logger.info(f'Set Software to {software}.')
+
+        self.update_units(units)
+
+        self.logger.info(f'Finding files with prefix: {prefix}, suffix: '
+                         f'{suffix} under directory {dir} produced by '
+                         f'{software}')
+        self.file_list = glob(dir + '/**/' + prefix + '*' + suffix,
+                         recursive=True)
+
+        self.logger.info(f'Found {len(self.file_list)} xvg files.')
+        self.logger.info("Unsorted file list: \n %s", '\n'.join(
+            self.file_list))
+
+        if software.lower() == 'gromacs':
+            self.logger.info(f'Using {software} parser to read the data.')
+            self._extract_u_nk = gmx.extract_u_nk
+            self._extract_dHdl = gmx.extract_dHdl
+        elif software.lower() == 'amber':
+            self._extract_u_nk = amber.extract_u_nk
+            self._extract_dHdl = amber.extract_dHdl
+        else: # pragma: no cover
+            raise NameError(f'{software} parser not found.')
+
+    def read(self):
+        '''Read the u_nk and dHdL data from the
+        :attr:`~alchemlyb.workflows.ABFE.file_list`
+
+        Attributes
+        ----------
+        u_nk_list : list
+            A list of :class:`pandas.DataFrame` of u_nk.
+        dHdl_list : list
+            A list of :class:`pandas.DataFrame` of dHdl.
+        '''
+        u_nk_list = []
+        dHdl_list = []
+        for xvg in self.file_list:
+            try:
+                u_nk = self._extract_u_nk(xvg, T=self.T)
+                self.logger.info(
+                    f'Reading {len(u_nk)} lines of u_nk from {xvg}')
+                u_nk_list.append(u_nk)
+            except: # pragma: no cover
+                self.logger.warning(
+                    f'Error reading read u_nk from {xvg}.')
```

Comment:
Why not die here immediately? It looks like a recipe for wrong/incomplete results. What's the rationale of only issuing a warning?

I would prefer that the workflow stops unless one explicitly sets something like `ignore_warnings=True` or something.